packages/plugin-ext/src/main/browser/tabs/tabs-main.ts

```diff
@@ -247,9 +272,11 @@ export class TabsMainImpl implements TabsMain, Disposable {
             tabDto,
             groupId: group.groupId
         });
+        this.waitQueue.get(args.title.owner)?.resolve();
```

Comment:
Wouldn't it make sense to remove the wait queue deferred immediately?

Reply:
We do, see 

https://github.com/eclipse-theia/theia/blob/c727f6d56a3b945b2d058a252cb88a2344368b55/packages/plugin-ext/src/main/browser/tabs/tabs-main.ts#L124-L127

Reply:
I've seen that, but is the "then" guaranteed to run in the same tick?

Reply:
No, but does it make any real difference?

Reply:
The only scenario I can see is that someone might invoke `waitForWidget` in the mean time, setting a new `deferred` into the `waitForQueue` and then the resolve would remove the second deferred. This would lead to no deferred in the map, the second deferred would get resolved after a second anyway, due to the timeout.

Reply:
I guess, it's unlikely, but I've adjusted the code 👍